### PR TITLE
Make `generativeModel` an instance method of `VertexAI`

### PR DESCRIPTION
**Pull request by andrewheard**
#no-changelog

### CODE REVIEW

File: FirebaseVertexAI/Sources/VertexAI.swift [hunk 1]
```diff
@@ -20,70 +20,81 @@ import Foundation
 @_implementationOnly import FirebaseCoreExtension
 
 @available(iOS 15.0, macOS 11.0, macCatalyst 15.0, *)
-@objc(FIRVertexAI)
```
Comment (paulb777) [inline FirebaseVertexAI/Sources/VertexAI.swift:23]: Does it need to be `objc` to be initialized by the Component system?

**Review by paulb777 — APPROVED** [review]
Update https://github.com/firebase/firebase-ios-sdk/blob/main/FirebaseCore/Sources/FIRApp.m#L828?

File: FirebaseVertexAI/Sources/VertexAI.swift (hunk 1 above)
Comment (andrewheard) [reply FirebaseVertexAI/Sources/VertexAI.swift:23]: I _think_ so. I was seeing `Cannot convert value of type '(any VertexAIProvider).Type' to expected argument type 'Protocol'` when I dropped the `@objc`s.

**Comment by andrewheard** [comment]
> Update [`main`/FirebaseCore/Sources/FIRApp.m#L828](https://github.com/firebase/firebase-ios-sdk/blob/main/FirebaseCore/Sources/FIRApp.m?rgh-link-date=2024-03-21T20%3A12%3A42Z#L828)?

@paulb777 Already set to `"fire-vtx"` in the `vertex-ai` branch (https://github.com/firebase/firebase-ios-sdk/commit/42df2e35933d0e38b2fa5bd3b27a01585bea9a18#diff-9c248fb2bb20f48431c46f7a4c98f41eadaeb637147b79772de10c443dde1edcR832) but could rename.

File: FirebaseVertexAI/Sources/VertexAI.swift (hunk 1 above)
Comment (paulb777) [reply FirebaseVertexAI/Sources/VertexAI.swift:23]: But it is dropped here?

**Pull request merged**